src/libs/common/ide/types/Messages.ts

```diff
@@ -15,7 +22,8 @@ export interface Messages {
    * @returns The option selected by the user, or `undefined` if no option was
    * selected
    */
-  showWarning(
+  showMessage(
```

Comment:
Made it so that the interface that each ide impl needs to conform to includes just one function, and then added utility wrappers `showWarning`, `showError`, etc